src/ralph/backends/data/base.py

```diff
@@ -74,7 +74,7 @@ def wrapper(*args, **kwargs):
         query = kwargs.pop("query", None)
         self_ = args[0]
 
-        yield from method(*args, query=self_.validate_query(query), **kwargs)
+        return method(*args, query=self_.validate_query(query), **kwargs)
```

Comment:
Not sure why you changed from `yield from` to `return`. Wouldn't this lose the traceback context? cf [here](https://discuss.python.org/t/difference-between-return-generator-vs-yield-from-generator/2997/5)